docs/features/traffic-management/plugins.md

```diff
@@ -0,0 +1,75 @@
+# Traffic Router Plugins
+
+!!! important Available since v1.5
+
+Argo Rollouts supports getting analysis metrics via 3rd party plugin system. This allows users to extend the capabilities of Rollouts
+to support metric providers that are not natively supported. Rollout's uses a plugin library called
+[go-plugin](https://github.com/hashicorp/go-plugin) to do this. You can find a sample plugin
+here: [sample-rollouts-trafficrouter-plugin](https://github.com/argoproj-labs/sample-rollouts-trafficrouter-plugin)
+
+## Using a Traffic Router Plugin
+
+There are two methods of installing and using an argo rollouts plugin. The first method is to mount up the plugin executable
+into the rollouts controller container. The second method is to use a HTTP(S) server to host the plugin executable.
+
+### Mounting the plugin executable into the rollouts controller container
+
+There are a few ways to mount the plugin executable into the rollouts controller container. Some of these will depend on your
+particular infrastructure. Here are a few methods:
+
+* Using an init container to download the plugin executable
+* Using a Kubernetes volume mount with a shared volume such as NFS, EBS, etc.
+* Building the plugin into the rollouts controller container
+
+Then you can use the configmap to point to the plugin executable file location. Example:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: argo-rollouts-config
+data:
+  plugins: |-
+    trafficrouters:
+    - plugin: "argoproj-labs/sample-rollouts-trafficrouter-plugin" # name of the plugin, it must match the name required by the plugin so it can find it's configuration
+      pluginLocation: "file://./my-custom-plugin" # supports http(s):// urls and file://
+```
+
+### Using a HTTP(S) server to host the plugin executable
+
+Argo Rollouts supports downloading the plugin executable from a HTTP(S) server. To use this method, you will need to
+configure the controller via the `argo-rollouts-config` configmap and set `pluginLocation` to a http(s) url. Example:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: argo-rollouts-config
+data:
+  plugins: |-
+    trafficrouters:
+    - plugin: "argoproj-labs/sample-rollouts-trafficrouter-plugin" # name of the plugin, it must match the name required by the plugin so it can find it's configuration
+      pluginLocation: "https://github.com/argoproj-labs/sample-rollouts-trafficrouter-plugin/releases/download/v0.0.3/metric-plugin-linux-amd64" # supports http(s):// urls and file://
+      pluginSha256: "08f588b1c799a37bbe8d0fc74cc1b1492dd70b2c" #optional sha256 checksum of the plugin executable
```

Comment:
```suggestion
    trafficrouters:
    - name: "argoproj-labs/sample-rollouts-trafficrouter-plugin" # name of the plugin, it must match the name required by the plugin so it can find it's configuration
      location: "https://github.com/argoproj-labs/sample-rollouts-trafficrouter-plugin/releases/download/v0.0.3/metric-plugin-linux-amd64" # supports http(s):// urls and file://
      sha256: "08f588b1c799a37bbe8d0fc74cc1b1492dd70b2c" #optional sha256 checksum of the plugin executable
```